### ISSUE TITLE
tokens count dont display in web ui

### ISSUE THREAD

**Issue by kareemalkoul** [Arize-ai/phoenix#1667]
i used phoenix with lama index
and i am create custom llm

```
from llama_index import ServiceContext, SimpleDirectoryReader, SummaryIndex
from llama_index.callbacks import CallbackManager
from llama_index.llms import (
    CustomLLM,
    CompletionResponse,
    CompletionResponseGen,
    LLMMetadata,
)
from llama_index.llms.base import llm_completion_callback


class OurLLM(CustomLLM):
    @property
    def metadata(self) -> LLMMetadata:
        """Get LLM metadata."""
        return LLMMetadata(
            context_window=4000,
            num_output=100,
            model_name="lama2-13B",
        )

    @llm_completion_callback()
    def complete(self, prompt: str, **kwargs: Any) -> CompletionResponse:
        (text, input_tokens, output_tokens, response) = sendPromptToLama(prompt)

        additional_kwargs={
                "prompt_tokens": input_tokens,
                "completion_tokens": output_tokens,
                "total_tokens": input_tokens + output_tokens,
            }
        print(additional_kwargs)
        return CompletionResponse(
            text=text,
            raw=response,
            additional_kwargs=additional_kwargs
        )

    @llm_completion_callback()
    def stream_complete(self, prompt: str, **kwargs: Any) -> CompletionResponseGen:
        raise NotImplementedError()

our_llm = OurLLM()

```
and the output in ui is
![image](https://github.com/Arize-ai/phoenix/assets/25280164/8bad2720-6cda-4d48-8918-f518e50df915)
so what is the problem cause this dont work token counts!

**Comment by mikeldking** [comment]
@kareemalkoul thanks for filing the issue! Looks like we are assuming 'usage" is attached to `raw` but it looks like this can also come from kwargs as you point out (https://github.com/run-llama/llama_index/blob/286f2f7964db061f455641490d429a7a0dcf4622/llama_index/llms/openai.py#L327). We will make the appropriate adjustments!

**Comment by kareemalkoul** [comment]
@mikeldking 
So if i add usage to raw(response) object then it will work ?

**Comment by mikeldking** [comment]
> @mikeldking So if i add usage to raw(response) object then it will work ?

@kareemalkoul it should if attached to the `raw.usage` - this is how OpenAI advertises the token usage I believe. But will add parsing for the `additional_kwargs` as well.

**Comment by kareemalkoul** [comment]
> > @mikeldking So if i add usage to raw(response) object then it will work ?
> 
> @kareemalkoul it should if attached to the `raw.usage` - this is how OpenAI advertises the token usage I believe. But will add parsing for the `additional_kwargs` as well.

then i do it 

````
from typing import Optional, List, Mapping, Any


class OurLLM(CustomLLM):
    @property
    def metadata(self) -> LLMMetadata:
        """Get LLM metadata."""
        return LLMMetadata(
            context_window=4000,
            num_output=100,
            model_name="lama2-13B",
        )

    @llm_completion_callback()
    def complete(self, prompt: str, **kwargs: Any) -> CompletionResponse:
        (text, input_tokens, output_tokens, response) = sendPromptToLama(prompt)

        additional_kwargs={
                "prompt_tokens": input_tokens,
                "completion_tokens": output_tokens,
                "total_tokens": input_tokens + output_tokens,
            }
        print(additional_kwargs)
        response["usage"]=additional_kwargs
        print(response)
        return CompletionResponse(
            text=text,
            raw=response,
            additional_kwargs=additional_kwargs
        )

    @llm_completion_callback()
    def stream_complete(self, prompt: str, **kwargs: Any) -> CompletionResponseGen:
        raise NotImplementedError()

our_llm = OurLLM()

```

but still dont work with token count!

**Comment by mikeldking** [comment]
> > > @mikeldking So if i add usage to raw(response) object then it will work ?
> > 
> > 
> > @kareemalkoul it should if attached to the `raw.usage` - this is how OpenAI advertises the token usage I believe. But will add parsing for the `additional_kwargs` as well.
> 
> then i do it
> 
> ```
> from typing import Optional, List, Mapping, Any
> 
> 
> class OurLLM(CustomLLM):
>     @property
>     def metadata(self) -> LLMMetadata:
>         """Get LLM metadata."""
>         return LLMMetadata(
>             context_window=4000,
>             num_output=100,
>             model_name="lama2-13B",
>         )
> 
>     @llm_completion_callback()
>     def complete(self, prompt: str, **kwargs: Any) -> CompletionResponse:
>         (text, input_tokens, output_tokens, response) = sendPromptToLama(prompt)
> 
>         additional_kwargs={
>                 "prompt_tokens": input_tokens,
>                 "completion_tokens": output_tokens,
>                 "total_tokens": input_tokens + output_tokens,
>             }
>         print(additional_kwargs)
>         response["usage"]=additional_kwargs
>         print(response)
>         return CompletionResponse(
>             text=text,
>             raw=response,
>             additional_kwargs=additional_kwargs
>         )
> 
>     @llm_completion_callback()
>     def stream_complete(self, prompt: str, **kwargs: Any) -> CompletionResponseGen:
>         raise NotImplementedError()
> 
> our_llm = OurLLM()
> 
> ```
> 
> but still dont work with token count! 
> ```
We assumed the `usage` to be not `Mapping` or `Dict` but we will add flexibility on our end! Will be in our next release. Thanks again.

**Comment by mikeldking** [comment]
@kareemalkoul Should be resolved in `arize-phoenix==0.0.50` - let us know how it goes!
